### PR TITLE
Mpeg: Add warm up staff

### LINKED ISSUE #13527: [Windows][Invalid Memory Access (Break)]GOD EATER & GOD EATER BURST (ULJS00237)(NPJH50352)(ULJS00350)
### What happens?
Turn off "Ignore Invalid Memory Access Error" check box.
These games start,
BANDAI&namco logos appeared and screenshot appear and stop.

And,Not response any key exept "Fast Fowarding Key".
If you click "Fast Fowarding Key" then the intro movie starting.

Ofcourse, If turn on the "Ignore Invalid Memory Access Error" chekc box then error screen is not appear.

### What *should* happen?
Work correctly.(Don't show the error screen.)

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.

PPSSPP Ver.：1.10.3-876
OS：Windows10（64-bit）Japanese
CPU：Ryzen7 3800X（8-core）
GPU：RADEON RX5700XT
Resolution：2560×1440
RAM：32GB

GODEATER Seriese_InvalidMemoryAccess

### PR DESCRIPTION
**Pull request by sum2012**
fix #13527

original log: https://gist.github.com/sum2012/ec213b7141b0fa08f6ae1a206faa92c7
modify log: https://gist.github.com/sum2012/1737d874a299a230a4ead995ff9eb556
Unlucky I cannot do JPCSPTrace of the game.Nothing record.
I do not sure which one need call ERROR_MPEG_NO_DATA

### CODE REVIEW

**Comment by sum2012** [comment]
The idea is from @unknownbrackets from very long ago

**Comment by hrydgard** [comment]
Oh, this looks plausible! Well, it'll probably not be exact but a probably quite reasonable approximation, and seems fairly safe.

Have you tested it a lot with other games?

**Comment by sum2012** [comment]
Megaman_Powered Up
don't have warm up  stream from JPCSPtrace.
https://gist.github.com/sum2012/e3adfd7fe890c6cf2ce37dd970c0d47c

Grand Theft Auto - Vice City Stories  required sceMpegGetAtracAu ERROR_MPEG_NO_DATA once

**Comment by sum2012** [comment]
I tested these 15 games

File: Core/HLE/sceMpeg.cpp [hunk 1]
```diff
@@ -1572,6 +1578,12 @@ static int sceMpegGetAvcAu(u32 mpeg, u32 streamId, u32 auAddr, u32 attrAddr)
 		return -1;
 	}
 
+	if (ctx->mpegwarmUp < MPEG_WARMUP_FRAMES) {
+		DEBUG_LOG(ME, "sceMpegGetAvcAu(%08x, %08x, %08x, %08x): warming uo", mpeg, streamId, auAddr, attrAddr);
```
Comment (shenweip) [inline Core/HLE/sceMpeg.cpp:1582]: minor typo.

File: Core/HLE/sceMpeg.cpp [hunk 2]
```diff
@@ -1667,6 +1679,12 @@ static int sceMpegGetAtracAu(u32 mpeg, u32 streamId, u32 auAddr, u32 attrAddr)
 		return -1;
 	}
 
+	if (ctx->mpegwarmUp < MPEG_WARMUP_FRAMES) {
+		DEBUG_LOG(ME, "sceMpegGetAtracAu(%08x, %08x, %08x, %08x): warning up", mpeg, streamId, auAddr, attrAddr);
```
Comment (shenweip) [inline Core/HLE/sceMpeg.cpp:1683]: minor typo.

**Comment by sum2012** [comment]
GOD EATER & GOD EATER BURST use custom mpeg.
I agree use compat.ini or detect don't normally load mpeg library to use warm up staff. 
 
33:35:908 HLE\sceKernelModule.cpp:1435 I[LOADER]: Exporting ent 196 named __usr__local__psp__devkit__lib__mpeg_stub_weak_a, 22 funcs, 0 vars, resident 089e86a4

**Comment by sum2012** [comment]
Might help #11060.

**Comment by hrydgard** [comment]
Let's try this .. just on the edge of too risky for 1.11 maybe. Can always revert.

**Comment by sum2012** [comment]
Test  on TALES OF PHANTASIA  ULJS00079 which mpeg library 0x104 
Also don't have warm up.
https://gist.github.com/sum2012/408d7364ff1d0a0a02470bec44876686

The video don't auto close to return menu within v1.10.3
Don't direct cause in this commit.
I will open a new issue when I get more information

**Pull request merged**